curvefs/src/mds/fs_info_wrapper.cpp

```diff
@@ -75,15 +75,32 @@ FsInfoWrapper::FsInfoWrapper(const ::curvefs::mds::CreateFsRequest* request,
     fsInfo_ = std::move(fsInfo);
 }
 
-bool FsInfoWrapper::IsMountPointExist(const Mountpoint& mp) const {
+bool FsInfoWrapper::IsMountPointExist(const Mountpoint &mp) const {
     return std::find_if(fsInfo_.mountpoints().begin(),
                         fsInfo_.mountpoints().end(),
-                        [mp](const Mountpoint& mountPoint) {
+                        [mp](const Mountpoint &mountPoint) {
                             return mp.path() == mountPoint.path() &&
                                    mp.hostname() == mountPoint.hostname();
                         }) != fsInfo_.mountpoints().end();
 }
 
+bool FsInfoWrapper::IsMountPointConflict(const Mountpoint &mp) const {
+    bool cto = (fsInfo_.mountpoints_size() ? false : mp.cto());
+
+    bool exist =
+        std::find_if(fsInfo_.mountpoints().begin(), fsInfo_.mountpoints().end(),
+                     [&](const Mountpoint &mountPoint) {
+                         if (mountPoint.has_cto() && mountPoint.cto()) {
+                             cto = true;
+                         }
+
+                         return mp.path() == mountPoint.path() &&
+                                mp.hostname() == mountPoint.hostname();
+                     }) != fsInfo_.mountpoints().end();
+
+    return exist || (!cto && mp.cto()) || (cto && !mp.cto());
```

Comment:
Here is not more clear and can add some code comments.

Reply:
fix

Reply:
so can use "return exist || cto != mp.cto()" instead ?

Reply:
fix